salt/addons/deploy_addons.sh

```diff
@@ -26,9 +26,19 @@ fi
 
 echo "Deploying DNS on Kubernetes"
 
+KUBEDNS_SA=`eval "$KUBECTL_PATH $NS_ARGS $SRV_ARGS get sa | grep kube-dns | cat"`
```

Comment:
I think we should try to move all this stuff from the `deploy_addons.sh` script to some state file, and just invoke `kubectl` with `cmd.run`.